compiler/src/iree/compiler/Dialect/HAL/Conversion/StreamToHAL/Patterns.cpp

```diff
@@ -106,14 +106,19 @@ struct ResourceAllocOpPattern
       return failure();
     }
 
+    auto memoryTypeOp =
+        rewriter.create<IREE::HAL::MemoryTypeOp>(allocOp.getLoc(), memoryTypes);
+    auto bufferUsageOp = rewriter.create<IREE::HAL::BufferUsageOp>(
+        allocOp.getLoc(), bufferUsage);
```

Comment:
```suggestion
    Value memoryTypesValue =
        rewriter.create<IREE::HAL::MemoryTypeOp>(allocOp.getLoc(), memoryTypes);
    Value bufferUsageValue = rewriter.create<IREE::HAL::BufferUsageOp>(
        allocOp.getLoc(), bufferUsage);
```

Any time you don't need the op you should make these Value so the implicit single result magic kicks in and you don't need the `getResult()` on each use.

(here and below)